consensus/src/state_replication.rs

```diff
@@ -47,14 +47,17 @@ pub trait StateComputer: Send + Sync {
         // The block that will be computed.
         block: &Block<Self::Payload>,
         // The executed trees of parent block.
-        executed_trees: ExecutedTrees,
-    ) -> Pin<Box<dyn Future<Output = Result<ProcessedVMOutput>> + Send>>;
+        parent_executed_trees: &ExecutedTrees,
+        // The last committed trees.
+        committed_trees: &ExecutedTrees,
```

Comment:
This is tricky. It may need data in storage that can be reached from `committed_trees`' version and verified by its state_root. It is self-explainable in state_view code.